integration_tests/tests/integrity/integrity_customers.sql

```diff
@@ -0,0 +1,35 @@
+{{ config(
+    tags="fivetran_validations",
+    enabled=var('fivetran_validation_tests_enabled', false)
+) }}
+
+with source as (
+    select
+        customer_id,
+        count(*) as source_customer_tag_count
+    from {{ target.schema }}_shopify_dev.stg_shopify__customer_tag
+    group by 1
+),
+
+transform as (
+    select
+        customer_id,
+        count(*) as transform_customer_tag_count
+    from {{ target.schema }}_shopify_dev.shopify__customers
+    where customer_tags is not null
+    group by 1
```

Comment:
This integrity test produces inaccurate results. This will count the number of tags in the source (accurate); however, it's only counting the number of rows for a given customer in the transform (inaccurate since this should always be 1). Therefore, this will result in a failed test if a given customer has multiple tags associated with them.

![image](https://github.com/user-attachments/assets/6c839597-e991-4a8d-ae38-8a6e9bb8aee9)

Instead we can adjust the transformation component of this test to check the items in the array and then count those. This way if a customer has 3 tags in the source, then there should be 3 items in the customer_tags array in the transform.

```suggestion
    select
        customer_id,
        array_length(split(customer_tags, ',')) as transform_customer_tag_count -- Only BigQuery compatible for the time being
    from zz_dbt_joe_shopify_dev.shopify__customers
    where customer_tags is not null
    group by customer_id, customer_tags
```

Reply:
Thank you. I have updated.